overlord/snapstate/snapstate_test.go

```diff
@@ -209,6 +210,40 @@ func verifyRemoveTasks(c *C, ts *state.TaskSet) {
 	})
 }
 
+func (s *snapmgrTestSuite) TestGenerateCookies(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// verify that GenerateCookies creates a cookie for a snap that's missing it.
+	s.state.Set("snaps", map[string]*json.RawMessage{
+		"some-snap":  nil,
+		"other-snap": nil})
+
+	// some-snap doesn't have cookie
+	contexts := map[string]string{"other-snap": "123456"}
+	s.state.Set("snap-cookies", contexts)
+
+	s.snapmgr.GenerateCookies(s.state)
+
+	err := s.state.Get("snap-cookies", &contexts)
+	c.Assert(err, IsNil)
+	c.Assert(contexts, HasLen, 2)
+
+	var cookie, snapName string
+	for cookie, snapName = range contexts {
```

Comment:
I couldn't quite do that since the map is indexed by cookie value.. However you made me realize I'm reading the cookie file a few lines below, so I can just reorder these checks a little bit and get rid of this loop. Thanks :)